### PR TITLE
Updating storybook docs 

### LINKED ISSUE #12618: Move storybook contribution guidelines to .storybook folder
**Describe the bug**
Move storybook contribution guidelines to .storybook folder as its more appropriate.

- Resolve or update `.storybook/README.md` to align with new docs
- Move `ui/1.INTRODUCTION.stories.mdx` and `ui/2.DOCUMENTATION.stories.mdx` to `.storybook` directory

### PR DESCRIPTION
**Pull request by georgewrmarshall**
Fixes: #12618

Explanation: Updating and moving docs to .storybook/ folder and updating the Button component. This will be useful for the upcoming IA Nav work and pointing contributors to an example of docs standard we want to strive for.

Manual testing steps:  
  -  Go to latest CI build of storybook
  -  Go to Documentation Guidelines and see updated section
  -  Go to "Button" component story and check controls work
  - Go to "Button" component docs page and see updated docs

**Images**

docs page

Screen Shot 2021-12-12 at 4 08 22 PM

btn docs

### CODE REVIEW

File: .storybook/2.DOCUMENTATION.stories.mdx [hunk 1]
```diff
@@ -14,18 +14,17 @@ Some general documentation best practices to follow:
 
 - Put yourself in the shoes of another developer trying to use the component you just created for the first time
 - Write a brief description of the component and what it's used for in the `README.mdx` file
-- Display the component's api using the `<ArgsTable of={YourComponent} />` component from storybook docs
-- Use the [controls](https://storybook.js.org/addons/@storybook/addon-controls/) api over knobs
-- Use the updated `argType` [actions](https://storybook.js.org/docs/react/essentials/actions) api over importing the actions plugin directly
-- Check the accessibility of the component using the **Accessibility** tab
+- Display the component's api using the `<ArgsTable of={YourComponent} />` component from storybook docs. Add descriptions of each prop by using jsDoc style comments in the `propTypes`.
```
Comment (darkwing) [inline .storybook/2.DOCUMENTATION.stories.mdx:17]: Tiny nit but I think we should uppercase `API`.  Please don't hurt me.

Comment (georgewrmarshall) [reply .storybook/2.DOCUMENTATION.stories.mdx]: Agreed! We should have a consistent terminology and standard. Have updated all instance of lowercase "api" to "API"

**Comment by BboyAkers** [comment]
This looks solid! 🙂

**Review by brad-decker — APPROVED** [review]

**Review by adonesky1 — APPROVED** [review]
LGTM!

**Pull request merged**